### PR TITLE
[aon_timer] Initial commit of AON Timer docs

### PR DESCRIPTION
**Pull request by tomroberts-lowrisc**
Just a basic translation of the proposal doc into markdown, plus an
initial draft of the HJSON spec.

Signed-off-by: Tom Roberts <tomroberts@lowrisc.org>

### CODE REVIEW

File: hw/ip/aon_timer/doc/_index.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+Since this timer functions as a watchdog, it has 2 additional functions when compared to the always on wake timer:
+* Watchdog configuration ‘Lock’
+* Watchdog count in sleep
+
+Unlike the always on timer, the watchdog once configured, may need to be locked by firmware until the next system reset.
+This is to prevent firmware from accidentally or maliciously disabling the watchdog.
+
+The “count in sleep” option controls whether the watchdog timer continues to count in low-power mode.
+This allows configurations where the watchdog timer can remain programmed and locked while the device is put to sleep for relatively long periods, controlled by the wakeup timer.
+Without this feature, the watchdog timer might wake up the core prematurely by triggering a watchdog bite.
```
Comment (rswarbrick) [inline hw/ip/aon_timer/doc/_index.md:56]: I don't think I understand this. Is the point that disabling counting in sleep allows the core to sleep for a while? And wouldn't the bark wake the core earlier? (Probably still to be avoided, but maybe the watchdog is biting prematurely!)

File: hw/ip/aon_timer/doc/_index.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+# Theory of Operations
+
+## Block Diagram
+
+![AON Timer Block Diagram](aon_timer_block_diagram.svg)
```
Comment (rswarbrick) [inline hw/ip/aon_timer/doc/_index.md:62]: Maybe it would be clearer if the counts and thresholds in the block diagram (`mtime`, `mtimecmp0`, `mtimecmp1`) were named after the corresponding CSRs? For example, `AON_WAKE` would use `WKUP_COUNT` and `WKUP_THOLD`.

File: hw/ip/aon_timer/doc/_index.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+See the block diagram for high level function and connectivity.
+The timer interacts with the CPU core and the pwrmgr/rstmgr to drive wakeup / reset events and interrupts.
```
Comment (rswarbrick) [inline hw/ip/aon_timer/doc/_index.md:65]: Probably better to expand "pwrmgr" and "rstmgr" to "power manager" and "reset manager" to be consistent with the text above (and the names that those blocks get in the documentation TOC).

File: hw/ip/aon_timer/doc/_index.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+There is also an extra input to tell the counter whether to run (“counter-run”).
+This is used to stop the watchdog timer running when in debugging mode or when the alert handler has put the system in a “killed” state.
```
Comment (rswarbrick) [inline hw/ip/aon_timer/doc/_index.md:67]: Non-ASCII quotes

File: hw/ip/aon_timer/doc/_index.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+## Hardware Interfaces
+
+{{< hwcfg "hw/ip/aon_timer/data/aon_timer.hjson" >}}
+
+## Design Details
+
+The always on timer will run on a ~200KHz clock.
+The timers themselves are 32b wide, giving a maximum timeout window of roughly ~6 hours.
+
+Note however, even though the timer core runs on a slow clock, register values are sampled into the main clock domain to ensure register read / writes do not incur large latencies.
```
Comment (rswarbrick) [inline hw/ip/aon_timer/doc/_index.md:78]: This sentence feels a bit clunky. I think we could just lose the first two words, which would help.

File: hw/ip/aon_timer/doc/_index.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+# Programmers Guide
+
+## Initialization
+
+It is recommended that software write the timer value {{<regref"WKUP_COUNT">}} and {{<regref"WDOG_COUNT">}} to zero before enabling either timer.
+The desired thresholds can then be programmed in {{<regref"WKUP_THOLD">}}, {{<regref"WDOG_BARK_THOLD">}} and {{<regref"WDOG_BITE_THOLD">}}.
+To enable the timers and start the count, set the enable bit to 1 in the {{<regref"WKUP_CTRL">}} / {{<regref"WDOG_CTRL">}} registers.
+Finally, if desired, the watchdog configuration can be locked by writing 1 to the regwen bit in {{<regref"WDOG_REGWEN">}}.
```
Comment (rswarbrick) [inline hw/ip/aon_timer/doc/_index.md:87]: The passive/active/direct/indirect moods here aren't consistent. Since the section is part of "Programmers Guide", I think we should just give instructions in the active mood (like the 3rd line). Also, I think this could be read as an instruction to start the timer "at the end of line 1" and before programming the threshold. Maybe just instruct the programmer to reset the counts, then set the thresholds, then enable the counters?

File: hw/ip/aon_timer/doc/_index.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+## Watchdog pet
+
+The watchdog can be petted by writing zero to the {{<regref"WDOG_COUNT">}} register.
```
Comment (rswarbrick) [inline hw/ip/aon_timer/doc/_index.md:91]: Again, I think this is easier to read as an instruction: "To pet the watchdog, write zero ..."

File: hw/ip/aon_timer/doc/_index.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+## Interrupt Handling
+
+If either timer reaches the programmed threshold, interrupts are generated from the AON_TIMER module.
+Clearing the interrupt is performed by writing 1 into the Interrupt Status register {{<regref "INTR_STATE">}}.
```
Comment (rswarbrick) [inline hw/ip/aon_timer/doc/_index.md:96]: And, again, this could be an instruction: "To clear the interrupt, write 1 into ..."

**Review by rswarbrick — COMMENTED** [review]
Hi Tom! I've got some nitpicky copy-edit comments, but the overall structure looks good to me.

File: hw/ip/aon_timer/doc/_index.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
+---
+title: "AON Timer Technical Specification"
+---
+
+# Overview
+
+This document specifies the Always-On ("AON") Timer IP functionality.
+This module conforms to the [Comportable guideline for peripheral functionality.]({{< relref "doc/rm/comportability_specification" >}})
+See that document for integration overview within the broader top level system.
```
[the remark this answers is not in the document]
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:9]: Done

File: hw/ip/aon_timer/doc/_index.md [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+
+## Features
+
+- Two 32-bit upcounting timers
+- One timer functions as a wakeup timer, one as a watchdog timer
+- Watchdog timer has two thresholds: bark (generates an interrupt) and bite (resets core)
```
[the remark this answers is not in the document]
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:15]: Done

File: hw/ip/aon_timer/doc/_index.md [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+
+## Description
+
+### AON Wakeup timer
+
+The always on wakeup timer operation is straightforward.
```
[the remark this answers is not in the document]
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:21]: Done

File: hw/ip/aon_timer/doc/_index.md [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+A count starts at 0 and slowly ticks upwards.
+When it reaches / exceeds the wake count, a level wakeup signal is sent to the power manager and a level IRQ is generated to the processor.
+This wakeup signal stays high until it is explicitly acknowledged by software.
+The IRQ behaves as all other IRQs in the system.
```
[the remark this answers is not in the document]
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:25]: Removed

File: hw/ip/aon_timer/doc/_index.md [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+The wakeup timer can be used like a real-time clock for long periods of low-power mode (though it does not give any guarantees of time-accuracy TODO: specify accuracy).
+
+### AON Watchdog timer
+
+The always on watchdog timer behaves similarly to the wakeup timer.
+A separate independent count starts at 0 and slowly ticks upwards.
```
[the remark this answers is not in the document]
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:31]: Done

File: hw/ip/aon_timer/doc/_index.md [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+When the first threshold is met or exceeded, a level wakeup signal (if enabled) is generated to the power manager.
+Simultaneously, a level IRQ signal is also generated to the processor.
+
+If the system is already in low power, the wakeup signal requests the power manager to wake the system such that the IRQ can be serviced.
```
[the remark this answers is not in the document]
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:35]: Done

File: hw/ip/aon_timer/doc/_index.md [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+If the system is not in low power mode, the IRQ is immediately serviced.
+Both the wakeup and the IRQ signals remain asserted until system reset or explicit acknowledgement by software.
+This first threshold is known as the watchdog bark.
+
+When the second threshold is met (this is known as the watchdog bite), a reset request is sent to the power manager which directly reboots the entire system.
+This is independent of the IRQ sent as part of the watchdog bark.
+The reset wipes out both the reset request and any pending IRQ requests, so software is not required to directly acknowledge them.
```
[the remark this answers is not in the document]
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:42]: It means the system reset - I'll rephrase it to make the point that the timer is also reset by the system reset, so nothing remains pending

File: hw/ip/aon_timer/doc/_index.md [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+
+To prevent the watchdog bark or bite, software is expected to periodically reset the count when opeating normally.
```
[the remark this answers is not in the document]
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:44]: Done

File: hw/ip/aon_timer/doc/_index.md [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+This is referred to as petting the watchdog, and is achieved by resetting the count to zero.
+
+Since this timer functions as a watchdog, it has 2 additional functions when compared to the always on wake timer:
```
[the remark this answers is not in the document]
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:47]: Done

File: hw/ip/aon_timer/doc/_index.md [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+* Watchdog configuration ‘Lock’
```
[the remark this answers is not in the document]
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:48]: Done

File: hw/ip/aon_timer/doc/_index.md [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+* Watchdog count in sleep
+
+Unlike the always on timer, the watchdog once configured, may need to be locked by firmware until the next system reset.
```
[the remark this answers is not in the document]
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:51]: Rephrased

File: hw/ip/aon_timer/doc/_index.md [hunk 20]
```diff
@@ -0,0 +1,104 @@
...
+This is to prevent firmware from accidentally or maliciously disabling the watchdog.
+
+The “count in sleep” option controls whether the watchdog timer continues to count in low-power mode.
```
[the remark this answers is not in the document]
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:54]: Done

File: hw/ip/aon_timer/doc/_index.md (hunk 1 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:56]: Should have said bark not bite! Also switched the name to "pause in sleep" to make it clear that the feature is to stop it from counting, and the default is for it to count.

File: hw/ip/aon_timer/doc/_index.md (hunk 2 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:62]: Done

File: hw/ip/aon_timer/doc/_index.md (hunk 3 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:65]: Done

File: hw/ip/aon_timer/doc/_index.md (hunk 4 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:67]: Done

File: hw/ip/aon_timer/doc/_index.md (hunk 5 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:78]: Rephrased

File: hw/ip/aon_timer/doc/_index.md (hunk 6 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:87]: Had another go at simplifying this

File: hw/ip/aon_timer/doc/_index.md (hunk 7 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:91]: Done

File: hw/ip/aon_timer/doc/_index.md (hunk 8 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:96]: Done

**Review by tomroberts-lowrisc — COMMENTED** [review]
Thanks for the review @rswarbrick

File: hw/ip/aon_timer/doc/_index.md [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+A count starts at 0 and slowly ticks upwards.
+When it reaches / exceeds the wake count, a level wakeup signal is sent to the power manager and a level IRQ is generated to the processor.
+This wakeup signal stays high until it is explicitly acknowledged by software.
+The wakeup timer can be used like a real-time clock for long periods of low-power mode (though it does not give any guarantees of time-accuracy TODO: specify accuracy).
+
+### AON Watchdog timer
+
+The always-on watchdog timer behaves similarly to the wakeup timer.
+It has an independent count starting at 0 which slowly ticks upwards.
+When the first threshold is met or exceeded, a level wakeup signal (if enabled) is generated to the power manager.
```
Comment (rswarbrick) [inline hw/ip/aon_timer/doc/_index.md:30]: Maybe "sent to the power manager"? (Or "generated and sent to", but I'm not sure that's needed for a wire going high!)

File: hw/ip/aon_timer/doc/_index.md [hunk 22]
```diff
@@ -0,0 +1,102 @@
...
+The system reset also resets the always-on timer, so software is not required to directly acknowledge anything after a watchdog reset.
+
+To prevent the watchdog bark or bite, software is expected to periodically reset the count when operating normally.
+This is referred to as petting the watchdog, and is achieved by resetting the count to zero.
+
+Since this timer functions as a watchdog, it has two additional functions not present in the always-on wakeup timer:
+* Watchdog configuration lock
+* Watchdog pause in sleep
+
+Unlike the wakeup timer, the watchdog timer configuration might need to be locked by firmware until the next system reset.
```
Comment (rswarbrick) [inline hw/ip/aon_timer/doc/_index.md:49]: Re-reading this, maybe just "... configuration can be locked by firmware ..." ? (Describing whether the firmware needs to do something is probably out of the scope of this document :-) )

**Review by rswarbrick — APPROVED** [review]
Thanks, Tom, that looks way better! A couple of other nits that I noticed when going through again, but this looks good.

File: hw/ip/aon_timer/doc/_index.md [hunk 23]
```diff
@@ -0,0 +1,102 @@
...
+
+# Theory of Operations
+
+## Block Diagram
+
+![AON Timer Block Diagram](aon_timer_block_diagram.svg)
+
+See the block diagram for high level function and connectivity.
+The timer interacts with the CPU core and the power manager and reset manager to drive wakeup / reset events and interrupts.
+There is also an extra input to tell the counter whether to run ("counter-run").
```
Comment (tjaychen) [inline hw/ip/aon_timer/doc/_index.md:64]: @msfschaffner 
there is chance this counter-run will actually turn into an escalation input (since that's how the life cycle communicates with everyone). We could sync up quickly on how this should work.

The life cycle actually broadcasts a lot of these escalation signals right now, so it might just be a matter of picking the right one to hook-up.

File: hw/ip/aon_timer/doc/_index.md [hunk 24]
```diff
@@ -0,0 +1,102 @@
...
+# Overview
+
+This document specifies the Always-On ("AON") Timer IP functionality.
+This module conforms to the [Comportable guideline for peripheral functionality.]({{< relref "doc/rm/comportability_specification" >}})
+See that document for an overview of how it is integrated into the top level system.
+
+## Features
+
+- Two 32-bit upcounting timers: one timer functions as a wakeup timer, one as a watchdog timer
+- The watchdog timer has two thresholds: bark (generates an interrupt) and bite (resets core)
```
Comment (tjaychen) [inline hw/ip/aon_timer/doc/_index.md:14]: do we need to make any clarifications about this not hooking up to NMI?
especially with the function of counting during low power?

File: hw/ip/aon_timer/doc/_index.md [hunk 25]
```diff
@@ -0,0 +1,102 @@
...
+
+The "pause in sleep" option controls whether the watchdog timer continues to count in low-power mode.
+This allows configurations where the watchdog timer can remain programmed and locked while the device is put to sleep for relatively long periods, controlled by the wakeup timer.
+Without this feature, the watchdog timer might wake up the core prematurely by triggering a watchdog bark.
+
+# Theory of Operations
+
+## Block Diagram
+
+![AON Timer Block Diagram](aon_timer_block_diagram.svg)
```
Comment (silvestrst) [inline hw/ip/aon_timer/doc/_index.md:60]: I think you have annotated bite and bark thresholds the wrong way around. I guess bite triggers the reset via rstmgr, right?

**Review by silvestrst — COMMENTED** [review]
@tomroberts-lowrisc thanks!

I have potentially found some "typo" kind of mistake in the diagram?

File: hw/ip/aon_timer/doc/_index.md [hunk 26]
```diff
@@ -0,0 +1,102 @@
...
+It has an independent count starting at 0 which slowly ticks upwards.
+When the first threshold is met or exceeded, a level wakeup signal (if enabled) is generated to the power manager.
+Simultaneously, a level IRQ signal is also generated to the processor.
+
+If the system is in a low power state, the wakeup signal asks the power manager to wake the system such that the IRQ can be serviced.
+If the system is not in low power mode, the IRQ is immediately serviced.
+Both the wakeup and the IRQ signals remain asserted until system reset or explicit acknowledgement by software.
+This first threshold is known as the watchdog bark.
+
+When the second threshold is met (this is known as the watchdog bite), a reset request is sent to the power manager which will reset the whole system.
```
Comment (silvestrst) [inline hw/ip/aon_timer/doc/_index.md:38]: Is the reset request guaranteed to succeed? Maybe it is worth word it more strongly, something like:
```
When the second threshold is met (this is known as the watchdog bite), a reset request is sent to the power manager which is guaranteed to trigger a System Reset.
```
Not sure what is the best way of wording it, but to me this sentence doesn't sound like the reset request is always guaranteed to succeed, even though it has `will` in the sentence.

Also, it seems that `reset the whole system` is maybe not a correct term to use here, as later on you mention the system reset - which is not whole system (POR would be whole system)?

File: hw/ip/aon_timer/doc/_index.md (hunk 25 above)
Comment (silvestrst) [inline hw/ip/aon_timer/doc/_index.md:60]: Am I reading the diagram correctly that the Wake-up timer interrupt goes through PLIC the same way watchdog bark does?

File: hw/ip/aon_timer/data/aon_timer.hjson [hunk 27]
```diff
@@ -0,0 +1,172 @@
...
+      swaccess: "wo",
+      hwaccess: "hro",
+      fields: [
+        { bits: "0",
+          name: "wkup_timer_expired",
+          desc: "Write 1 to force wkup_timer_expired interrupt",
+        }
+        { bits: "1",
+          name: "wdog_timer_expired",
+          desc: "Write 1 to force wdog_timer_expired interrupt",
```
Comment (silvestrst) [inline hw/ip/aon_timer/data/aon_timer.hjson:166]: I think you should add that this is `bark` interrupt the same you have done in the `INTR_STATE` register description.

File: hw/ip/aon_timer/doc/_index.md (hunk 22 above)
Comment (silvestrst) [inline hw/ip/aon_timer/doc/_index.md:49]: Could you please specify what registers/functionality is locked by this bit (we still have to pet the watchdog even if the lock bit is set).

File: hw/ip/aon_timer/doc/_index.md [hunk 28]
```diff
@@ -0,0 +1,102 @@
...
+This is referred to as petting the watchdog, and is achieved by resetting the count to zero.
+
+Since this timer functions as a watchdog, it has two additional functions not present in the always-on wakeup timer:
+* Watchdog configuration lock
+* Watchdog pause in sleep
+
+Unlike the wakeup timer, the watchdog timer configuration might need to be locked by firmware until the next system reset.
+This allows the option of preventing firmware from accidentally or maliciously disabling the watchdog.
+
+The "pause in sleep" option controls whether the watchdog timer continues to count in low-power mode.
```
Comment (silvestrst) [inline hw/ip/aon_timer/doc/_index.md:52]: `low-power mode` -> `low-power modes`? There are "Sleep" and "Deep Sleep" low power modes.

File: hw/ip/aon_timer/doc/_index.md (hunk 22 above)
Comment (silvestrst) [reply hw/ip/aon_timer/doc/_index.md:49]: Just realised that every affected register has `regwen` field in hjson.

File: hw/ip/aon_timer/doc/_index.md [hunk 29]
```diff
@@ -0,0 +1,102 @@
...
+# Programmers Guide
+
+## Initialization
+
+1. Write the timer values {{<regref"WKUP_COUNT">}} and {{<regref"WDOG_COUNT">}} to zero.
+2. Program the desired thresholds in {{<regref"WKUP_THOLD">}}, {{<regref"WDOG_BARK_THOLD">}} and {{<regref"WDOG_BITE_THOLD">}}.
+3. Set the enable bit to 1 in the {{<regref"WKUP_CTRL">}} / {{<regref"WDOG_CTRL">}} registers.
+4. If desired, lock the watchdog configuration by writing 1 to the `regwen` bit in {{<regref"WDOG_REGWEN">}}.
+
+## Watchdog pet
```
Comment (silvestrst) [inline hw/ip/aon_timer/doc/_index.md:87]: How about the wake-up timer, does it have to be cleared, or is it cleared automatically?

**Comment by tjaychen** [comment]
@tomroberts-lowrisc one of the customer teams had a look, and they were wondering if you could add some information regarding the pre-scaler for cases where they want a REALLY long lived count?

File: hw/ip/aon_timer/doc/_index.md [hunk 30]
```diff
@@ -0,0 +1,102 @@
...
+There is also an extra input to tell the counter whether to run ("counter-run").
+This is used to stop the watchdog timer running when in debugging mode or when the alert handler has put the system in a "killed" state.
+
+## Hardware Interfaces
+
+{{< hwcfg "hw/ip/aon_timer/data/aon_timer.hjson" >}}
+
+## Design Details
+
+The always-on timer will run on a ~200KHz clock.
```
Comment (tjaychen) [inline hw/ip/aon_timer/doc/_index.md:73]: @rswarbrick 
do you guys think it makes sense for this kind of information to link to the [ast documentation](https://github.com/lowRISC/opentitan/blob/master/hw/top_earlgrey/ip/ast/doc/_index.md)?

Comment (rswarbrick) [reply hw/ip/aon_timer/doc/_index.md:73]: Hmm, maybe. It depends how self-contained and modular we want the blocks to be. All the comportability stuff seems to suggest we want to be able to grab IP blocks and slap them down in other systems. If we're doing that, then "upward" links like that in the documentation are probably confusing. Of course, in reality, we'll probably never do it (and will invent completely new tooling for the next major project, because one always does...), in which case we should totally link everything together. So... not sure!

File: hw/ip/aon_timer/data/aon_timer.hjson (hunk 27 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/data/aon_timer.hjson:166]: Will do

File: hw/ip/aon_timer/doc/_index.md (hunk 24 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:14]: Not sure what you mean here, can you clarify?

File: hw/ip/aon_timer/doc/_index.md (hunk 21 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:30]: Will fix

File: hw/ip/aon_timer/doc/_index.md (hunk 26 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:38]: Agree `whole system` doesn't make sense, will change to system reset. Not sure I want to say it's guaranteed, since it's just a request (which the power manager theoretically could ignore).

File: hw/ip/aon_timer/doc/_index.md (hunk 22 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:49]: Agree will change

File: hw/ip/aon_timer/doc/_index.md (hunk 28 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:52]: thanks - will update

File: hw/ip/aon_timer/doc/_index.md (hunk 25 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:60]: Yep! will fix

Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:60]: yes

File: hw/ip/aon_timer/doc/_index.md (hunk 30 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:73]: It would be useful (for software) to specify things like the clock precision to know what the error bars look like on timeouts. I don't think any of that kind of information is (or ever will be) in the AST documentation though.

File: hw/ip/aon_timer/doc/_index.md (hunk 29 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:87]: Not sure what you mean here - the whole purpose of the timer is to run in low-power and wake up the system.

File: hw/ip/aon_timer/doc/_index.md (hunk 23 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:64]: That's what I was hoping - is there a standard way of specifying that hook-up in the hjson?

**Review by tomroberts-lowrisc — COMMENTED** [review]
thanks for the review @tjaychen and @silvestrst 

I left out the prescaler stuff from the document as I wasn't sure if we wanted it. I'll add it back in shortly.

File: hw/ip/aon_timer/doc/_index.md (hunk 29 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:87]: Ah maybe you mean: once it's triggered and woken the system up it will keep on counting (and triggering interrupts). I'll add a hint to the interrupt handling section to disable or reinitialize the counter once it has triggered.

File: hw/ip/aon_timer/doc/_index.md (hunk 26 above)
Comment (silvestrst) [reply hw/ip/aon_timer/doc/_index.md:38]: > Not sure I want to say it's guaranteed, since it's just a request (which the power manager theoretically could ignore).

Makes sense, do you know a potential use-case when such request may get ignored by the Power Manager? Seems slightly counter-intuitive if watchdog event is dropped.

File: hw/ip/aon_timer/doc/_index.md (hunk 25 above)
Comment (silvestrst) [reply hw/ip/aon_timer/doc/_index.md:60]: Thanks, the formatting in the diagram looked as if the behaviour there is different (on through PLIC, the other one not).

File: hw/ip/aon_timer/doc/_index.md (hunk 29 above)
Comment (silvestrst) [reply hw/ip/aon_timer/doc/_index.md:87]: > Ah maybe you mean: once it's triggered and woken the system up it will keep on counting (and triggering interrupts).

Yes. Didn't quite explain myself well in the previous comment.

File: hw/ip/aon_timer/doc/_index.md (hunk 25 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:60]: I'll just add the PLIC into the diagram to make it clearer.

File: hw/ip/aon_timer/doc/_index.md (hunk 26 above)
Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:38]: I guess it's possible that it would be ignored if the lifecycle escalation has put the system in a killed state (forcing the user to do the reset)

File: hw/ip/aon_timer/doc/_index.md (hunk 23 above)
Comment (tjaychen) [reply hw/ip/aon_timer/doc/_index.md:65]: it's the same way as using inter-module to do hook-ups between modules.
However, the exact signal to use is standardized.   Mine i think are all in the review pipeline right now, so i'll circle back with an example once they merge.

File: hw/ip/aon_timer/doc/_index.md (hunk 24 above)
Comment (tjaychen) [reply hw/ip/aon_timer/doc/_index.md:14]: o just that the interrupt should not be hooked up to NMI at a system level.
Since that 'could' potentially cause ROM to get stuck due to this module being always on.

Comment (tomroberts-lowrisc) [reply hw/ip/aon_timer/doc/_index.md:14]: Gotcha - I'll add a note
